packages/flutter_tools/lib/src/localizations/gen_l10n.dart

```diff
@@ -123,23 +123,36 @@ String _syntheticL10nPackagePath(FileSystem fileSystem) => fileSystem.path.join(
 // For example, if placeholders are used for plurals and no type was specified, then the type will
 // automatically set to 'num'. Similarly, if such placeholders are used for selects, then the type
 // will be set to 'String'. For such placeholders that are used for both, we should throw an error.
-List<String> generateMethodParameters(Message message, bool useNamedParameters) {
-  return message.placeholders.values.map((Placeholder placeholder) {
+List<String> generateMethodParameters(Message message, LocaleInfo? locale, bool useNamedParameters) {
+
+  // check the compatibility of template placeholders and locale placeholders.
+  final Map<String, Placeholder>? localePlaceholders = message.localePlaceholders[locale];
+
+  return message.templatePlaceholders.entries.map((MapEntry<String, Placeholder> e) {
+    final Placeholder placeholder = e.value;
+    final String? localePlaceholderType = localePlaceholders?[e.key]?.type;
+    if (localePlaceholderType != null && placeholder.type != localePlaceholderType) {
```

Comment:
I would like to make the type explicit in this pull request. I think it would be better to loosen the type constraint in another pull request if necessary.

Reply:
Right, i'm saying that I _want_ the type check to be strict, so I always want the type in the template ARB and the other ARBs to always match.

```suggestion
    if (placeholder.type != localePlaceholderType) {
```